packages/core/src/editor/BlockNoteEditor.ts

```diff
@@ -749,10 +863,18 @@ export class BlockNoteEditor<
     return this._tiptapEditor.view;
   }
 
+  /**
+   * The state of the editor when a transaction is captured, this can be continuously updated during the {@link transact} call
+   */
+  private transactionState: EditorState | null = null;
+
   /**
    * Get the underlying prosemirror state
    */
   public get prosemirrorState() {
+    if (this.transactionState) {
+      return this.transactionState;
```

Comment:
this can now be not in sync with the view. can this cause downstream issues you think?

Reply:
Another thing that could go wrong is calling `dispatch` and then trying to find the node in the view (dom), right? For example, getting an element by id. Previously this would have worked, so I suppose this is also something to check / be mindful of?